holidays/holiday_base.py

```diff
@@ -773,6 +773,16 @@ def _add_special_holidays(self):
                 ):
                     self._add_holiday(name, date(self._year, month, day))
 
+        if self.observed:
```

Comment:
This belongs to `ObservedHolidayBase`. You need to override `ObservedHolidayBase::_add_special_holidays`.